### PR TITLE
feat(samples): add CMakeLists.txt for mpi/ccl samples

### PR DESCRIPTION
**Pull request by homksei**

### CODE REVIEW

**Review by ethanglaser — COMMENTED** [review]
Nice work on this, but have a few questions that may be resolved by addition of a description:
Is just public CI LinuxMake_GNUMKL to be impacted by these changes? Are the same changes meant for Linux DPCPP public CI job as well or is that still to be make? Are oneapi/dpc ccl and mpi CMakeLists.txts used anywhere currently (ie any internal CI)?

File: samples/oneapi/dpc/ccl/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# limitations under the License.
+#===============================================================================
+
+cmake_minimum_required(VERSION 3.16)
+
+set(ONEDAL_USE_DPCPP yes)
+set(ONEDAL_INTERFACE yes)
+set(ONEDAL_DISTRIBUTED yes)
+set(ONEDAL_USE_CCL yes)
+set(MPIEXEC_MAX_NUMPROCS 1)
```
[the remark this answers is not in the document]
Comment (homksei) [reply samples/oneapi/dpc/ccl/CMakeLists.txt]: Yes, by default. 
Added the ability to override by passing `-DMPIEXEC_MAX_NUMPROCS=<num>`/`-DMPIEXEC_NUMPROCS_PER_NODE=<num>`.

**Comment by homksei** [comment]
> Nice work on this, but have a few questions that may be resolved by addition of a description:
> 
> Is just public CI LinuxMake_GNUMKL to be impacted by these changes? Are the same changes meant for Linux DPCPP public CI job as well or is that still to be make? Are oneapi/dpc ccl and mpi CMakeLists.txts used anywhere currently (ie any internal CI)?

mpi samples are launched only for LinuxMake_GNUMKL in public CI. If we decide to add it to the Linux DPCPP public CI job, they can also be launched using CMake. I would also suggest deprecating the ability to run mpi/ccl samples through make (and .bat in the case of Windows) and leave only CMake.

**Comment by ethanglaser** [comment]
> mpi samples are launched only for LinuxMake_GNUMKL in public CI. If we decide to add it to the Linux DPCPP public CI job, they can also be launched using CMake.

I think we are launching mpi samples in Linux DPCPP public CI step - https://github.com/oneapi-src/oneDAL/blob/main/.ci/pipeline/ci.yml#L204-L207

**Comment by homksei** [comment]
> > mpi samples are launched only for LinuxMake_GNUMKL in public CI. If we decide to add it to the Linux DPCPP public CI job, they can also be launched using CMake.
> 
> Looks like we are launching mpi samples in Linux DPCPP public CI step - https://github.com/oneapi-src/oneDAL/blob/main/.ci/pipeline/ci.yml#L204-L207


Thank you for bringing this to my attention. I completely forgot that it's enabled there. Switched to CMake there too.

**Review by ethanglaser — APPROVED** [review]

**Review by Alexandr-Solovev — APPROVED** [review]
Nice work!

**Comment by napetrov** [comment]
@homksei  @ethanglaser  @Alexandr-Solovev  - should we go ahead and integrate?

**Review by napetrov — APPROVED** [review]

**Pull request merged**